CI fails on g++5 Ubuntu 16.04 missing isinf

I observed this for all PRs since the updated Travis CI was merged. When `log2f_non_ieee` was added, we introduced a bug for Ubuntu 16.04 with G++ 5. Apparently it builds for gcc-5 for the C part but things fail on the C++ side.

On the one hand this a bug specific to Ubuntu 16.04/g++5 on the other hand, this is an LTS release. Arguably Ubuntu 16.04 is very old and you shouldn't use old compilers/distros with new libraries but we need to make a decision for this one how to handle this error.

1. Just drop the Ubuntu 16.04/g++5 test
2. Change CI to use Ubuntu 18.04/g++5 OR Ubuntu 14.04/g++5 (previous mode)
3. Fix this as a conditional bug with all kinds of possible side-effects.

[A similar bug report](https://github.com/facebookarchive/caffe2/issues/1633)

```
Scanning dependencies of target volk_test_all
[ 89%] Building CXX object lib/CMakeFiles/volk_test_all.dir/testqa.cc.o
In file included from /home/travis/build/gnuradio/volk/build/include/volk/volk.h:31:0,
                 from /home/travis/build/gnuradio/volk/lib/qa_utils.h:5,
                 from /home/travis/build/gnuradio/volk/lib/kernel_tests.h:1,
                 from /home/travis/build/gnuradio/volk/lib/testqa.cc:31:

/home/travis/build/gnuradio/volk/include/volk/volk_common.h: In function ‘float log2f_non_ieee(float)’:
/home/travis/build/gnuradio/volk/include/volk/volk_common.h:153:24: error: ‘isinf’ was not declared in this scope
     return isinf(result) ? copysignf(127.0f, result) : result;
                        ^

/home/travis/build/gnuradio/volk/include/volk/volk_common.h:153:24: note: suggested alternative:
In file included from /usr/include/c++/5/complex:44:0,
                 from /home/travis/build/gnuradio/volk/include/volk/volk_complex.h:23,
                 from /home/travis/build/gnuradio/volk/build/include/volk/volk_typedefs.h:30,
                 from /home/travis/build/gnuradio/volk/build/include/volk/volk.h:29,
                 from /home/travis/build/gnuradio/volk/lib/qa_utils.h:5,
                 from /home/travis/build/gnuradio/volk/lib/kernel_tests.h:1,
                 from /home/travis/build/gnuradio/volk/lib/testqa.cc:31:

/usr/include/c++/5/cmath:626:5: note:   ‘std::isinf’
     isinf(_Tp __x)
     ^

lib/CMakeFiles/volk_test_all.dir/build.make:62: recipe for target 'lib/CMakeFiles/volk_test_all.dir/testqa.cc.o' failed
make[2]: *** [lib/CMakeFiles/volk_test_all.dir/testqa.cc.o] Error 1
CMakeFiles/Makefile2:200: recipe for target 'lib/CMakeFiles/volk_test_all.dir/all' failed
make[1]: *** [lib/CMakeFiles/volk_test_all.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -26,26 +26,20 @@ matrix:
           - ${TRAVIS_BUILD_DIR}/cache
       before_script:
         - cd ${TRAVIS_BUILD_DIR} && ./scripts/ci/download_intel_sde.sh
-    
+
     # Job 2 ... gcc-4.8
     - name: Linux Ubuntu 14.04 (trusty) x86 GCC 4.8
       dist: trusty
       env: MATRIX_EVAL="CC=gcc-4.8 && CXX=g++-4.8"
       addons: {apt: {packages: [*common_packages, g++-4.8, libboost-system-dev, libboost-filesystem-dev]}}
 
-    # Job 3 ... gcc-5
-    - name: Linux Ubuntu 16.04 (xenial) x86 GCC 5
-      dist: xenial
-      env: MATRIX_EVAL="CC=gcc-5 && CXX=g++-5"
-      addons: {apt: {sources: "ubuntu-toolchain-r-test", packages: [*common_packages, g++-5]}}
-    
     # Job 4 ... gcc-6
     - name: Linux x86 GCC 6
       env: MATRIX_EVAL="CC=gcc-6 && CXX=g++-6"
       addons: {apt: {sources: "ubuntu-toolchain-r-test", packages: [*common_packages, g++-6]}}
-    
+
     # Job 5 ... gcc-7
-    - name: Linux x86 GCC 7 
+    - name: Linux x86 GCC 7
       env: MATRIX_EVAL="CC=gcc-7 && CXX=g++-7"
       addons: {apt: {sources: "ubuntu-toolchain-r-test", packages: [*common_packages, g++-7]}}
 
@@ -53,23 +47,23 @@ matrix:
     - name: Linux x86 GCC 8
       env: MATRIX_EVAL="CC=gcc-8 && CXX=g++-8"
       addons: {apt: {sources: "ubuntu-toolchain-r-test", packages: [*common_packages, g++-8]}}
-    
+
     # Job 7 ... ARMv7 cross compile
     - name: Linux ARMv7 Qemu GCC 7
       env: MATRIX_EVAL="CC=gcc-7 && CXX=g++-7 && CMAKE_ARG=-DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchains/arm-linux-gnueabihf.cmake"
       addons: {apt: {sources: "ubuntu-toolchain-r-test", packages: [*common_packages, g++-arm-linux-gnueabihf, qemu-user]}}
-    
+
     # Job 8 ... ARMv8 (aarch64) cross compile
     - name: Linux ARMv8 (aarch64) Qemu GCC 7
       env: MATRIX_EVAL="CC=gcc-7 && CXX=g++-7 && CMAKE_ARG=-DCMAKE_TOOLCHAIN_FILE=../cmake/Toolchains/aarch64-linux-gnu.cmake"
       addons: {apt: {sources: "ubuntu-toolchain-r-test", packages: [*common_packages, g++-aarch64-linux-gnu, qemu-user]}}
- 
+
     # Job 9 ... clang
     - name: Linux x86 Clang 6
       env: MATRIX_EVAL="CC=\"clang -fprofile-instr-generate -fcoverage-mapping\" && CXX=\"clang++ -fprofile-instr-generate -fcoverage-mapping\""
       addons: {apt: {packages: [*common_packages, ]}}
 
-    - name: Linux ARMv8 (aarch64) GCC 7 
+    - name: Linux ARMv8 (aarch64) GCC 7
       arch: arm64
       env: MATRIX_EVAL="CC=gcc-7 && CXX=g++-7"
       addons: {apt: {packages: [*common_packages, ]}}
@@ -106,9 +100,7 @@ matrix:
 script:
   - eval "${MATRIX_EVAL}"
   - lscpu
-  - mkdir build && cd build && BOOST_ROOT=$BOOST_ROOT 
-  - cmake ${CMAKE_ARG} ../ 
+  - mkdir build && cd build && BOOST_ROOT=$BOOST_ROOT
+  - cmake ${CMAKE_ARG} ../
   - make
   - ctest -V
-
-

